src/app/[locale]/components/TaskDetails.tsx

```diff
@@ -0,0 +1,326 @@
+"use client";
+import { useState, useEffect } from "react";
+import { type TaskData, type UserData } from "../types/typescript";
+import { useForm } from "react-hook-form";
+import { useTranslations } from "next-intl";
+import { useDispatch, useSelector } from "react-redux";
+import { selectAuth } from "../../../redux/slices/authSlice";
+import { AppDispatch } from "../../../redux/store";
+import {
+  getTask,
+  selectTask,
+  updateTask,
+} from "../../../redux/slices/taskSlice";
+import { getUsers, selectUser } from "../../../redux/slices/userSlice";
+import { useRouter } from "next/navigation";
+import { selectGroup } from "../../../redux/slices/groupSlice";
+
+export default function TaskDetails({
+  id,
+  locale,
+}: {
+  id: string;
+  locale: string;
+}) {
+  const [edit, setEdit] = useState<boolean>(false);
+  const { isLoggedIn } = useSelector(selectAuth);
+  const { task } = useSelector(selectTask);
+  const { users } = useSelector(selectUser);
+  const { groups } = useSelector(selectGroup);
+  const dispatch = useDispatch<AppDispatch>();
+  const [authorisedUser, setAuthorisedUser] = useState<boolean>(false);
+  const [admin, setAdmin] = useState<boolean>(false);
+  const t = useTranslations("translations");
+  const router = useRouter();
+  const [filteredUsers, setFilteredUsers] = useState<UserData[]>([]);
+
+  const enableEditMode = (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (!id) return false;
+    setEdit(true);
+  };
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+    watch,
+  } = useForm({
+    defaultValues: {
+      title: task?.title,
+      description: task?.description,
+      type: task?.type,
+      createdOn: task?.createdOn,
+      status: task?.status,
+      assignedTo: task?.assignedTo,
+      group: task?.group?._id,
+    },
+  });
+
+  const onSubmit = async (data: TaskData) => {
+    dispatch(updateTask(data));
+    setEdit(false);
+  };
+
+  useEffect(() => {
+    dispatch(getUsers());
+
+    if (
+      isLoggedIn?.roles.includes("manager") ??
```

Comment:
these can be changed into constants for readability

Reply:
fixed